### ISSUE TITLE
Things to Do

### ISSUE THREAD

**Issue by FeGeyer** [beckstev/MachineLearningSeminar#21]
- ~~Display predicted labels~~

- ~~use pretrained net (kaggle)~~, ~~Netzwerkoptimierung~~

- ~~alternativ-methode: feature map aus autoencoder und dann in RF~~

- ~~GridSearch as self-written function~~

**Comment by beckstev** [comment]
Good tutorial to build an autoencoder:
https://ramhiser.com/post/2018-05-14-autoencoders-with-keras/

**Comment by beckstev** [comment]
Final work:

- [x] Evaluation of hyper parameter testing 
- [x] Training network with optimal hyper parameters
- [x] Created a plot to show the input of the NN 
- [x] Comparision plot alternative method and NN (confusion matrix)
- [ ] Understanding Plot
- [ ] Comparision of difference archtitectures (impact of he_normal and PReLu)
- [x] Präsentation erstellen 
- [ ] Worst 5 Best 5

**Comment by beckstev** [comment]
:(
https://machinelearningmastery.com/reproducible-results-neural-networks-keras/

**Comment by beckstev** [comment]
@FritzGeise 
The loss and acc plots of the Pretrained models  can not be used for our final report, because we are currently using dropout without tracking the loss/acc after each iteration. As you may remember, Mr. Nackenhorst showed us a callback method to do this. We must implement this before the report is due.

**Comment by beckstev** [comment]
> @FritzGeise
The loss and acc plots of the Pretrained models can not be used for our final report, because we are currently using dropout without tracking the loss/acc after each iteration. As you may remember, Mr. Nackenhorst showed us a callback method to do this. We must implement this before the report is due.

Addressed in pull request https://github.com/beckstev/MachineLearningSeminar/pull/48

**Comment by beckstev** [comment]
### For the report
We could use in the dataloader a `dict` instead of this array: 
https://github.com/beckstev/MachineLearningSeminar/blob/e4112e02a92e8fbd27394f65b4210fcbcb1eed3a/dog_classifier/net/dataloader.py#L123-L124
This would give as the advantage to save all images in original size because the items of a dict __must not have__ the same size!
